mlos_bench/mlos_bench/services/remote/azure/azure_auth.py

```diff
@@ -47,12 +51,49 @@ def __init__(self,
         # Register methods that we want to expose to the Environment objects.
         self.register([self.get_access_token])
 
+        if "spClientId" in self.config:
+            check_required_params(
+                self.config, {
+                    "keyVaultName",
+                    "certName",
+                    "spClientId",
+                }
+            )
+
         # This parameter can come from command line as strings, so conversion is needed.
         self._req_interval = float(self.config.get("tokenRequestInterval", self._REQ_INTERVAL))
+        keyvault_name = self.config.get("keyVaultName")
+        cert_name = self.config.get("certName")
+        sp_client_id = self.config.get("spClientId")
+        tenant_id = self.config.get("tenant", self._TENTANT_ID)
 
         self._access_token = "RENEW *NOW*"
         self._token_expiration_ts = datetime.datetime.now()  # Typically, some future timestamp.
 
+        # Login as ourselves
+        local_user_cred = azure_id.AzureCliCredential()
+        self._cred = local_user_cred
```

Comment:
Why not just
```suggestion
        self._cred = azure_id.AzureCliCredential()
```
and use `self._cred` everywhere? pylint might be complaining about the number of local variables already